scipy/optimize/_basinhopping.py

```diff
@@ -310,7 +312,18 @@ def accept_reject(self, energy_new, energy_old):
         If new is higher than old, there is a chance it will be accepted,
         less likely for larger differences.
         """
-        w = math.exp(min(0, -float(energy_new - energy_old) * self.beta))
+        with warnings.catch_warnings():
+            # The energy values being fed to Metropolis are 1-length arrays, and if
+            # they are equal, their difference is 0, which gets multiplied by beta,
+            # which is inf, and array([0]) * float('inf') causes
+            #
+            # RuntimeWarning: invalid value encountered in multiply
+            #
+            # Ignore this warning so so when the algorithm is on a flat plane, it always
+            # accepts the step, to try to move off the plane.
+            warnings.simplefilter("ignore", category=RuntimeWarning)
+            prod = -(energy_new - energy_old) * self.beta
+        w = math.exp(min(0, prod))
```

Comment:
Rather than catch the warning, you could maybe change the error flags to ignore it in the first place:
```
old = np.seterr(invalid='ignore')
prod = -(energy_new - energy_old) * self.beta
np.seterr(**old)
```

Reply:
`with np.errstate(invalid='ignore'): ...`

Reply:
Done.